agent_api/src/main/java/dev/aikido/agent_api/storage/ServiceConfiguration.java

```diff
@@ -64,21 +71,86 @@ public void setMiddlewareInstalled(boolean middlewareInstalled) {
         this.middlewareInstalled = middlewareInstalled;
     }
 
-    public HashSet<String> getBypassedIPs() {
-        return bypassedIPs;
-    }
-
     public List<Endpoint> getEndpoints() {
         return endpoints;
     }
 
-    public HashSet<String> getBlockedUserIDs() {
-        return blockedUserIDs;
+    public boolean isUserBlocked(String userId) {
+        return this.blockedUserIDs.contains(userId);
+    }
+
+    public boolean isIpBypassed(String ip) {
+        return this.bypassedIPs.matches(ip);
+    }
+
+    /**
+     * Check if the IP is blocked (e.g. Geo IP Restrictions)
+     */
+    public BlockedResult isIpBlocked(String ip) {
+        // Check for allowed ip addresses (i.e. only one country is allowed to visit the site)
+        // Always allow access from private IP addresses (those include local IP addresses)
+        if (!allowedIps.isEmpty() && !isPrivateIp(ip)) {
+            boolean ipAllowed = false;
+            for (IPListEntry entry : allowedIps) {
+                if (entry.ipList.matches(ip)) {
+                    ipAllowed = true; // We allow IP addresses as long as they match with one of the lists.
+                    break;
+                }
+            }
+            if (!ipAllowed) {
+                return new BlockedResult(true, "not in allowlist");
+            }
+        }
+
+        // Check for blocked ip addresses
+        for (IPListEntry entry : blockedIps) {
+            if (entry.ipList.matches(ip)) {
+                return new BlockedResult(true, entry.description);
+            }
+        }
+        return new BlockedResult(false, null);
+    }
+
+    public void updateBlockedLists(ReportingApi.APIListsResponse res) {
+        // clear
+        blockedIps.clear();
+        allowedIps.clear();
+        blockedUserAgentRegex = null;
```

Comment:
Would maybe move these closer to each if? And the comment doesn't add much imo

Reply:
was mostly to add description to diff steps in function, but moved them to each individual if statement